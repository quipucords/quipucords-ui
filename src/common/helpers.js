import moment from 'moment';
import _get from 'lodash/get';
import _set from 'lodash/set';

const copyClipboard = text => {
  let successful;

  try {
    window.getSelection().removeAllRanges();

    const newTextarea = window.document.createElement('pre');
    newTextarea.appendChild(window.document.createTextNode(text));

    newTextarea.style.position = 'absolute';
    newTextarea.style.top = '-9999px';
    newTextarea.style.left = '-9999px';

    const range = window.document.createRange();
    window.document.body.appendChild(newTextarea);

    range.selectNode(newTextarea);

    window.getSelection().addRange(range);

    successful = window.document.execCommand('copy');

    window.document.body.removeChild(newTextarea);
    window.getSelection().removeAllRanges();
  } catch (e) {
    successful = null;

    if (process.env.REACT_APP_ENV !== 'test') {
      console.warn('Copy to clipboard failed.', e.message);
    }
  }

  return successful;
};

const devModeNormalizeCount = (count, modulus = 100) => Math.abs(count) % modulus;

const downloadData = (data = '', fileName = 'download.txt', fileType = 'text/plain') =>
  new Promise((resolve, reject) => {
    try {
      const blob = new Blob([data], { type: fileType });

      if (window.navigator && window.navigator.msSaveBlob) {
        window.navigator.msSaveBlob(blob, fileName);
        resolve({ fileName, data });
      } else {
        const anchorTag = window.document.createElement('a');

        anchorTag.href = window.URL.createObjectURL(blob);
        anchorTag.style.display = 'none';
        anchorTag.download = fileName;

        window.document.body.appendChild(anchorTag);

        anchorTag.click();

        setTimeout(() => {
          window.document.body.removeChild(anchorTag);
          window.URL.revokeObjectURL(blob);
          resolve({ fileName, data });
        }, 250);
      }
    } catch (error) {
      reject(error);
    }
  });

const generateId = prefix =>
  `${prefix || 'generatedid'}-${(process.env.REACT_APP_ENV !== 'test' && Math.ceil(1e5 * Math.random())) || ''}`;

const noop = Function.prototype;

const noopPromise = Promise.resolve({});

const noopTranslate = (key, value) => value || `t('${key}')`;

const sourceTypeIcon = sourceType => {
  switch (sourceType) {
    case 'vcenter':
      return { type: 'pf', name: 'vcenter' };
    case 'network':
      return { type: 'pf', name: 'network-range' };
    case 'satellite':
      return { type: 'pf', name: 'satellite' };
    default:
      return { type: 'pf', name: '' };
  }
};

const scanTypeIcon = scanType => {
  switch (scanType) {
    case 'connect':
      return { type: 'pf', name: 'connected' };
    case 'inspect':
      return { type: 'fa', name: 'search' };
    default:
      return { type: 'pf', name: '' };
  }
};

const scanStatusIcon = scanStatus => {
  switch (scanStatus) {
    case 'completed':
    case 'success':
      return { type: 'pf', name: 'ok', classNames: [] };
    case 'failed':
    case 'canceled':
      return { type: 'pf', name: 'error-circle-o', classNames: [] };
    case 'unreachable':
      return { type: 'pf', name: 'disconnected', classNames: ['is-error'] };
    case 'created':
    case 'pending':
    case 'running':
      return { type: 'fa', name: 'spinner', classNames: ['fa-spin'] };
    case 'paused':
      return { type: 'pf', name: 'warning-triangle-o', classNames: [] };
    default:
      console.error(`Unknown status: ${scanStatus}`);
      return { type: 'pf', name: 'unknown', classNames: [] };
  }
};

const setPropIfDefined = (obj, props, value) => (obj && value !== undefined ? _set(obj, props, value) : obj);

const setPropIfTruthy = (obj, props, value) => (obj && value ? _set(obj, props, value) : obj);

const viewPropsChanged = (nextViewOptions, currentViewOptions) =>
  nextViewOptions.currentPage !== currentViewOptions.currentPage ||
  nextViewOptions.pageSize !== currentViewOptions.pageSize ||
  nextViewOptions.sortField !== currentViewOptions.sortField ||
  nextViewOptions.sortAscending !== currentViewOptions.sortAscending ||
  nextViewOptions.activeFilters !== currentViewOptions.activeFilters;

const createViewQueryObject = (viewOptions, queryObj) => {
  const queryObject = {
    ...queryObj
  };

  if (viewOptions) {
    if (viewOptions.sortField) {
      queryObject.ordering = viewOptions.sortAscending ? viewOptions.sortField : `-${viewOptions.sortField}`;
    }

    if (viewOptions.activeFilters) {
      viewOptions.activeFilters.forEach(filter => {
        queryObject[filter.field.id] = filter.field.filterType === 'select' ? filter.value.id : filter.value;
      });
    }

    queryObject.page = viewOptions.currentPage;
    queryObject.page_size = viewOptions.pageSize;
  }

  return queryObject;
};

const getMessageFromResults = (results, filter = null) => {
  const status = _get(results, 'response.status', results.status);
  const statusResponse = _get(results, 'response.statusText', results.statusText);
  const messageResponse = _get(results, 'response.data', results.message);
  const detailResponse = _get(results, 'response.data', results.detail);
  const requestUrl = _get(results, 'config.url', null);

  const messages = {
    messages: {},
    message: null,
    status: status || 0,
    url: requestUrl
  };

  const displayStatus = status >= 500 ? `${status} ` : '';

  if (!messageResponse && !detailResponse) {
    if (status < 400) {
      messages.message = statusResponse;
      return messages;
    }

    if (status >= 500 || status === undefined) {
      messages.message = `${status || ''} Server is currently unable to handle this request.`;
      return messages;
    }
  }

  if (status >= 500 && /Request\sURL:/.test(messageResponse)) {
    messages.message = `${status} ${messageResponse.split(/Request\sURL:/)[0]}`;
    return messages;
  }

  if (status === 404) {
    messages.message = `404 Not Found`;
    return messages;
  }

  if (typeof messageResponse === 'string') {
    messages.message = `${displayStatus}${messageResponse}`;
    return messages;
  }

  if (typeof detailResponse === 'string') {
    messages.message = `${displayStatus}${detailResponse}`;
    return messages;
  }

  const getMessages = messageObjectArrayString => {
    const parsed = {};

    if (messageObjectArrayString && typeof messageObjectArrayString === 'string') {
      return messageObjectArrayString.replace(/\[object\sObject]/g, '');
    }

    if (Array.isArray(messageObjectArrayString)) {
      return getMessages(messageObjectArrayString.join('\n'));
    }

    Object.keys(messageObjectArrayString).forEach(key => {
      parsed[key] = getMessages(messageObjectArrayString[key]);
    });

    return parsed;
  };

  const filtered = (messageObjectArrayString, filterField) => {
    const parsed = {};
    const str = JSON.stringify(messageObjectArrayString);
    const filterFields = (Array.isArray(filterField) && filterField) || (filterField && [filterField]) || [];

    filterFields.forEach(val => {
      const match = str.match(new RegExp(`"${val}":(\\[(.*?)]]?|{(.*?)}]?|"(.*?)"),?`));

      if (match && match[1]) {
        parsed[val] = match[1]; // eslint-disable-line

        if (match && match[2]) {
          parsed[val] = JSON.parse(match[1]);

          if (Array.isArray(parsed[val])) {
            parsed[val] = parsed[val].join(', ');
          } else if (typeof parsed[val] !== 'string') {
            parsed[val] = Object.values(parsed[val]).join(', ');
          }

          parsed[val] = parsed[val].replace(/,?\s?\[object\sObject]/, '');
        }
      }
    });

    return parsed;
  };

  messages.messages =
    (filter && filtered(messageResponse || detailResponse, filter)) || getMessages(messageResponse || detailResponse);
  messages.message = `${displayStatus}${Object.values(messages.messages).join('\n')}`;

  if (messages.message === '[object Object]') {
    messages.message = JSON.stringify(messages.messages);
  } else {
    messages.message = messages.message.replace(/\n?\[object\sObject]/g, '');
  }

  return messages;
};

const getStatusFromResults = results => {
  let status = _get(results, 'response.status', results.status);

  if (status === undefined) {
    status = 0;
  }

  return status;
};

const getTimeStampFromResults =
  process.env.REACT_APP_ENV !== 'test'
    ? results => moment(_get(results, 'headers.date', Date.now())).format('YYYYMMDD_HHmmss')
    : () => '20190225_164640';

const getTimeDisplayHowLongAgo =
  process.env.REACT_APP_ENV !== 'test'
    ? timestamp =>
        moment
          .utc(timestamp)
          .utcOffset(moment().utcOffset())
          .fromNow()
    : () => 'a day ago';

const isIpAddress = name => {
  const vals = name.split('.');
  if (vals.length === 4) {
    return vals.find(val => Number.isNaN(val)) === undefined;
  }
  return false;
};

const ipAddressValue = name => {
  const values = name.split('.');
  return values[0] * 0x1000000 + values[1] * 0x10000 + values[2] * 0x100 + values[3] * 1;
};

const DEV_MODE = process.env.REACT_APP_ENV === 'development';

const PROD_MODE = process.env.REACT_APP_ENV === 'production';

const TEST_MODE = process.env.REACT_APP_ENV === 'test';

const UI_BRAND = process.env.REACT_APP_UI_BRAND === 'true';

const UI_NAME = process.env.REACT_APP_UI_NAME;

const UI_SENTENCE_START_NAME = process.env.REACT_APP_UI_SENTENCE_START_NAME;

const UI_SHORT_NAME = process.env.REACT_APP_UI_SHORT_NAME;

const UI_VERSION = process.env.REACT_APP_UI_VERSION;

const helpers = {
  copyClipboard,
  devModeNormalizeCount,
  downloadData,
  generateId,
  noop,
  noopPromise,
  noopTranslate,
  sourceTypeIcon,
  scanTypeIcon,
  scanStatusIcon,
  setPropIfDefined,
  setPropIfTruthy,
  viewPropsChanged,
  createViewQueryObject,
  getMessageFromResults,
  getStatusFromResults,
  getTimeStampFromResults,
  getTimeDisplayHowLongAgo,
  isIpAddress,
  ipAddressValue,
  DEV_MODE,
  PROD_MODE,
  TEST_MODE,
  UI_BRAND,
  UI_NAME,
  UI_SENTENCE_START_NAME,
  UI_SHORT_NAME,
  UI_VERSION
};

export { helpers as default, helpers };
