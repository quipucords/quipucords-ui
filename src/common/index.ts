import { helpers } from './helpers';

export { helpers as default, helpers };
