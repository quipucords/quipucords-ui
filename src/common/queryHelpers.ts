import { TableState } from '@mturley-latest/react-table-batteries';
import { useQuery } from '@tanstack/react-query';
import axios from 'axios';
import helpers from 'src/common';

export const getServiceQueryUrl = <
  TItem,
  TColumnKey extends string,
  TSortableColumnKey extends TColumnKey
>({
  tableState: {
    filter: { filterValues },
    sort: { activeSort, initialSort },
    pagination: { pageNumber, itemsPerPage }
  },
  baseUrl,
  columnOrderMap
}: {
  tableState: TableState<TItem, TColumnKey, TSortableColumnKey>;
  baseUrl?: string;
  columnOrderMap?: Record<TSortableColumnKey, string>;
}) => {
  const filterParams = filterValues
    ? Object.keys(filterValues)
        .map(key => `${key}=${filterValues[key]}`)
        .join('&')
    : null;

  const ordering = `${(activeSort?.direction ?? initialSort?.direction) === 'desc' ? '-' : ''}${
    activeSort?.columnKey
      ? columnOrderMap?.[activeSort.columnKey] || activeSort.columnKey
      : initialSort?.columnKey
  }`;

  const query =
    `${baseUrl}` +
    `?` +
    `ordering=${ordering}` +
    `&` +
    `page=${pageNumber}` +
    `&` +
    `page_size=${itemsPerPage}${filterParams ? `&${filterParams}` : ''}`;

  return query;
};

export type ServiceQueryResult<TItem> = { count: number; results: TItem[] };

export const useServiceQuery = <
  TItem,
  TColumnKey extends string,
  TSortableColumnKey extends TColumnKey
>({
  queryKey,
  baseUrl,
  columnOrderMap,
  tableState,
  setRefreshTime
}: {
  queryKey: string[];
  baseUrl?: string;
  columnOrderMap?: Record<TSortableColumnKey, string>;
  tableState: TableState<TItem, TColumnKey, TSortableColumnKey>;
  setRefreshTime?: (date: Date) => void;
}) =>
  useQuery<ServiceQueryResult<TItem>>({
    queryKey: [...queryKey, tableState.cacheKey],
    refetchOnWindowFocus: !helpers.DEV_MODE,
    queryFn: async () => {
      try {
        const query = getServiceQueryUrl({ tableState, baseUrl, columnOrderMap });
        console.log(`Query: `, query);
        const response = await axios.get<ServiceQueryResult<TItem>>(query, {
          headers: { Authorization: `Token ${localStorage.getItem('authToken')}` }
        });
        setRefreshTime?.(new Date());
        return response.data;
      } catch (error) {
        console.error(error);
        throw error; // You can choose to throw the error or return a default value here
      }
    }
  });
