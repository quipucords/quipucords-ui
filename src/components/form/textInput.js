import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { TextInput as PfTextInput, TextInputTypes } from '@patternfly/react-core';
import { createMockEvent } from './formHelpers';
import { helpers } from '../../common';

/**
 * A wrapper for Patternfly TextInput.
 * Provides a consistent event structure, and an onClear event for the search type.
 *
 * @fires onKeyUp
 * @fires onMouseUp
 * @fires onChange
 * @param {object} props
 * @param {string} props.className
 * @param {string} props.id
 * @param {boolean} props.isDisabled
 * @param {string} props.name
 * @param {Function} props.onChange
 * @param {Function} props.onClear
 * @param {Function} props.onKeyUp
 * @param {Function} props.onMouseUp
 * @param {boolean} props.isReadOnly
 * @param {string} props.type
 * @param {*|string} props.value
 * @param {object} props.props
 * @returns {React.ReactNode}
 */
const TextInput = ({
  className,
  id,
  isDisabled,
  name,
  onChange,
  onClear,
  onKeyUp,
  onMouseUp,
  isReadOnly,
  type,
  value,
  ...props
}) => {
  const [updatedValue, setUpdatedValue] = useState(value);
  const [clearedEvent, setClearedEvent] = useState();
  const [changeEvent, setChangeEvent] = useState();

  useEffect(() => {
    if (clearedEvent) {
      setClearedEvent(null);
      onClear(clearedEvent);
    }
  }, [clearedEvent, onClear]);

  useEffect(() => {
    if (changeEvent) {
      setChangeEvent(null);
      onChange(changeEvent);
    }
  }, [changeEvent, onChange]);

  // ToDo: review having to manually set value on escape key
  /**
   * onKeyUp event, provide additional functionality for onClear event.
   *
   * @event onKeyUp
   * @param {object} event
   */
  const onTextInputKeyUp = event => {
    const { keyCode } = event;
    const clonedEvent = { ...event };

    onKeyUp(createMockEvent(event, true));

    if (keyCode === 27) {
      setUpdatedValue('');
      const updatedTarget = clonedEvent.currentTarget;
      updatedTarget.value = '';
      setClearedEvent(createMockEvent({ ...clonedEvent, currentTarget: updatedTarget }));
    }
  };

  // ToDo: review use of setTimeout, used for pf icons inside field
  /**
   * onMouseUp event, provide additional functionality for onClear event.
   *
   * @event onMouseUp
   * @param {object} event
   */
  const onTextInputMouseUp = event => {
    const { currentTarget } = event;
    const clonedEvent = { ...event };

    onMouseUp(createMockEvent(event, true));

    if (type !== 'search' || currentTarget.value === '') {
      return;
    }

    window.setTimeout(() => {
      if (currentTarget.value === '') {
        onClear(createMockEvent(clonedEvent));
      }
    });
  };

  /**
   * onChange event, provide restructured event.
   *
   * @event onChange
   * @param {string} changedValue
   * @param {object} event
   */
  const onTextInputChange = (changedValue, event) => {
    const clonedEvent = { ...event };

    setUpdatedValue(changedValue);
    setChangeEvent(createMockEvent(clonedEvent));
  };

  const updatedName = name || helpers.generateId();
  const updatedId = id || updatedName;

  return (
    <PfTextInput
      id={updatedId}
      name={updatedName}
      className={`quipucords-form__text-input ${className}`}
      isDisabled={isDisabled || false}
      isReadOnly={isReadOnly || false}
      onChange={onTextInputChange}
      onKeyUp={onTextInputKeyUp}
      onMouseUp={onTextInputMouseUp}
      type={type}
      value={updatedValue ?? value ?? ''}
      {...props}
    />
  );
};

/**
 * Prop types
 *
 * @type {{onKeyUp: Function, isReadOnly: boolean, onChange: Function, onClear: Function, name: string,
 *     className: string, id: string, isDisabled: boolean, onMouseUp: Function, type: string,
 *     value: string}}
 */
TextInput.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  isDisabled: PropTypes.bool,
  isReadOnly: PropTypes.bool,
  name: PropTypes.string,
  onChange: PropTypes.func,
  onClear: PropTypes.func,
  onKeyUp: PropTypes.func,
  onMouseUp: PropTypes.func,
  type: PropTypes.oneOf([...Object.values(TextInputTypes)]),
  value: PropTypes.string
};

/**
 * Default props
 *
 * @type {{onKeyUp: Function, isReadOnly: boolean, onChange: Function, onClear: Function, name: null,
 *     className: string, id: null, isDisabled: boolean, onMouseUp: Function, type: TextInputTypes.text,
 *     value: string}}
 */
TextInput.defaultProps = {
  className: '',
  id: null,
  isDisabled: false,
  isReadOnly: false,
  name: null,
  onChange: helpers.noop,
  onClear: helpers.noop,
  onKeyUp: helpers.noop,
  onMouseUp: helpers.noop,
  type: TextInputTypes.text,
  value: ''
};

export { TextInput as default, TextInput, TextInputTypes };
