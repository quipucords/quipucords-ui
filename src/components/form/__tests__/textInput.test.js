import React from 'react';
import { TextInput as PfTextInput, TextInputTypes } from '@patternfly/react-core';
import { TextInput } from '../textInput';
import { helpers } from '../../../common';

describe('TextInput Component', () => {
  it('should render a basic component', async () => {
    const props = {};

    const component = await shallowHookComponent(<TextInput {...props} />);
    expect(component.render()).toMatchSnapshot('basic component');
  });

  it('should export support constants, types', () => {
    expect({ TextInputTypes }).toMatchSnapshot('support');
  });

  it('should handle readOnly, disabled', async () => {
    const props = {
      isReadOnly: true
    };

    const component = await mountHookComponent(<TextInput {...props} />);
    expect(component.render()).toMatchSnapshot('readOnly');

    component.setProps({
      isReadOnly: false,
      isDisabled: true
    });

    expect(component.render()).toMatchSnapshot('disabled');

    component.setProps({
      isReadOnly: false,
      isDisabled: false
    });

    expect(component.render()).toMatchSnapshot('active');
  });

  it('should return an emulated onChange event', async () => {
    const mockOnChange = jest.fn();
    const props = {
      onChange: mockOnChange,
      value: 'lorem ipsum'
    };

    const component = await mountHookComponent(<TextInput {...props} />);
    const mockEvent = { currentTarget: { value: 'dolor sit' }, persist: helpers.noop };
    component.find(PfTextInput).simulate('change', 'hello world', mockEvent);

    expect(mockOnChange.mock.calls).toMatchSnapshot('emulated event, change');
  });

  it('should return an emulated onClear event on escape', async () => {
    const mockOnClear = jest.fn();
    const props = {
      onClear: mockOnClear,
      value: 'lorem ipsum'
    };

    const component = await mountHookComponent(<TextInput {...props} />);
    const mockEvent = { keyCode: 27, currentTarget: { value: '' }, persist: helpers.noop };
    component.find(PfTextInput).simulate('keyup', mockEvent);

    expect(mockOnClear.mock.calls).toMatchSnapshot('emulated event, esc');
  });

  it('should return an emulated onClear event on escape with type search', async () => {
    const mockOnClear = jest.fn();
    const props = {
      onClear: mockOnClear,
      value: 'lorem ipsum',
      type: 'search'
    };

    const component = await mountHookComponent(<TextInput {...props} />);
    const mockEvent = { keyCode: 27, currentTarget: { value: '' }, persist: helpers.noop };
    component.find(PfTextInput).simulate('keyup', mockEvent);

    expect(mockOnClear.mock.calls).toMatchSnapshot('emulated event, esc, type search');
  });

  it('should return a mouseup event on text clear', async () => {
    const mockOnMouseUp = jest.fn();
    const props = {
      onMouseUp: mockOnMouseUp,
      value: 'lorem ipsum'
    };

    const component = await shallowHookComponent(<TextInput {...props} />);
    const mockEvent = { currentTarget: { value: '' }, persist: helpers.noop };
    component.find(PfTextInput).simulate('mouseup', mockEvent);

    expect(mockOnMouseUp.mock.calls).toMatchSnapshot('emulated event, mouseup');
  });
});
