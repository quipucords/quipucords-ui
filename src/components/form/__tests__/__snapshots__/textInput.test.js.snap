// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component should export support constants, types: support 1`] = `
Object {
  "TextInputTypes": Object {
    "date": "date",
    "datetimeLocal": "datetime-local",
    "email": "email",
    "month": "month",
    "number": "number",
    "password": "password",
    "search": "search",
    "tel": "tel",
    "text": "text",
    "time": "time",
    "url": "url",
  },
}
`;

exports[`TextInput Component should handle readOnly, disabled: active 1`] = `
<input
  aria-invalid="false"
  class="pf-c-form-control quipucords-form__text-input "
  data-ouia-component-id="OUIA-Generated-TextInputBase-2"
  data-ouia-component-type="PF4/TextInput"
  data-ouia-safe="true"
  id="generatedid-"
  name="generatedid-"
  type="text"
  value=""
/>
`;

exports[`TextInput Component should handle readOnly, disabled: disabled 1`] = `
<input
  aria-invalid="false"
  class="pf-c-form-control quipucords-form__text-input "
  data-ouia-component-id="OUIA-Generated-TextInputBase-2"
  data-ouia-component-type="PF4/TextInput"
  data-ouia-safe="true"
  disabled=""
  id="generatedid-"
  name="generatedid-"
  type="text"
  value=""
/>
`;

exports[`TextInput Component should handle readOnly, disabled: readOnly 1`] = `
<input
  aria-invalid="false"
  class="pf-c-form-control quipucords-form__text-input "
  data-ouia-component-id="OUIA-Generated-TextInputBase-2"
  data-ouia-component-type="PF4/TextInput"
  data-ouia-safe="true"
  id="generatedid-"
  name="generatedid-"
  readonly=""
  type="text"
  value=""
/>
`;

exports[`TextInput Component should render a basic component: basic component 1`] = `
<input
  aria-invalid="false"
  class="pf-c-form-control quipucords-form__text-input "
  data-ouia-component-id="OUIA-Generated-TextInputBase-1"
  data-ouia-component-type="PF4/TextInput"
  data-ouia-safe="true"
  id="generatedid-"
  name="generatedid-"
  type="text"
  value=""
/>
`;

exports[`TextInput Component should return a mouseup event on text clear: emulated event, mouseup 1`] = `
Array [
  Array [
    Object {
      "checked": undefined,
      "currentTarget": Object {
        "value": "",
      },
      "id": undefined,
      "keyCode": undefined,
      "name": undefined,
      "persist": [Function],
      "target": Object {},
      "value": "",
    },
  ],
]
`;

exports[`TextInput Component should return an emulated onChange event: emulated event, change 1`] = `
Array [
  Array [
    Object {
      "checked": undefined,
      "currentTarget": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-3"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="generatedid-"
        name="generatedid-"
        type="text"
        value="lorem ipsum"
      />,
      "id": "generatedid-",
      "keyCode": undefined,
      "name": "generatedid-",
      "persist": [Function],
      "target": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-3"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="generatedid-"
        name="generatedid-"
        type="text"
        value="lorem ipsum"
      />,
      "value": "lorem ipsum",
    },
  ],
]
`;

exports[`TextInput Component should return an emulated onClear event on escape with type search: emulated event, esc, type search 1`] = `
Array [
  Array [
    Object {
      "checked": undefined,
      "currentTarget": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-5"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="generatedid-"
        name="generatedid-"
        type="search"
        value=""
      />,
      "id": "generatedid-",
      "keyCode": 27,
      "name": "generatedid-",
      "persist": [Function],
      "target": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-5"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="generatedid-"
        name="generatedid-"
        type="search"
        value=""
      />,
      "value": "",
    },
  ],
]
`;

exports[`TextInput Component should return an emulated onClear event on escape: emulated event, esc 1`] = `
Array [
  Array [
    Object {
      "checked": undefined,
      "currentTarget": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-4"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="generatedid-"
        name="generatedid-"
        type="text"
        value=""
      />,
      "id": "generatedid-",
      "keyCode": 27,
      "name": "generatedid-",
      "persist": [Function],
      "target": <input
        aria-invalid="false"
        class="pf-c-form-control quipucords-form__text-input "
        data-ouia-component-id="OUIA-Generated-TextInputBase-4"
        data-ouia-component-type="PF4/TextInput"
        data-ouia-safe="true"
        id="generatedid-"
        name="generatedid-"
        type="text"
        value=""
      />,
      "value": "",
    },
  ],
]
`;
